packages/gatsby/src/utils/node-manifest.ts

```diff
@@ -320,37 +295,43 @@ export async function processNodeManifest(
  * and then removes them from the store.
  * Manifest files are added via the public unstable_createNodeManifest action in sourceNodes
  */
-export async function processNodeManifests(): Promise<void> {
+export async function processNodeManifests(): Promise<Record<string, string>> {
+  const startTime = Date.now()
   const { nodeManifests } = store.getState()
 
   const totalManifests = nodeManifests.length
 
   if (totalManifests === 0) {
-    return
+    return {}
   }
 
-  const processedManifests = await Promise.all(
-    nodeManifests.map(manifest => processNodeManifest(manifest))
-  )
-
   let totalProcessedManifests = 0
   let totalFailedManifests = 0
-
-  processedManifests.forEach(manifest => {
-    if (manifest) {
-      totalProcessedManifests++
-    } else {
-      totalFailedManifests++
-    }
-  })
+  const nodeManifestPagePathMap = {}
+
+  await Promise.all(
+    nodeManifests.map(async manifest => {
+      const processedManifest = await processNodeManifest(manifest)
+
+      if (processedManifest && processedManifest?.page?.path) {
+        nodeManifestPagePathMap[processedManifest.page.path] =
```

Comment:
This is where we can use `cache.set` instead of storing an object

Reply:
Come to think of it if we shoudl also encapsulate this logic inside of `processNodeManifest`

Reply:
You mean instead returning this
```
  const finalManifest: INodeManifestOut = {
    node: inputManifest.node,
    page: nodeManifestPage,
    foundPageBy,
  }
```
To just return what we actually care about: `{ "pagePath": "manifestId" }`?

Reply:
Yep exactly :D, we don't need to return cache.set but we do want to await it

Reply:
Don't we need to return a promise though?

Reply:
An async function always returns a promise by default :) iirc `return cache.set` would be a promise returning another promise

Reply:
Ok I think I'm wrong about that, it wont return another promise, but we don't need the return value so for readability it's better not to return it imo

Reply:
Ah gotcha, I did not realize that making a function `async` made it return a promise by default!

Reply:
Async await is syntactic sugar on top of promises 😄 if your async fn doesn't return a promise it will wrap the return value in a promise